relayer/processor/path_end_runtime.go

```diff
@@ -316,6 +316,30 @@ func (pathEnd *pathEndRuntime) shouldSendPacketMessage(message packetIBCMessage,
 		)
 		return false
 	}
+
+	removeRetention := func() {
+		// giving up on sending this packet flow message
+		// remove all retention of this connection handshake in pathEnd.messagesCache.PacketFlow and counterparty
+		toDelete := make(map[string][]uint64)
+		toDeleteCounterparty := make(map[string][]uint64)
+		switch eventType {
+		case chantypes.EventTypeRecvPacket:
+			toDelete[eventType] = []uint64{sequence}
+			toDeleteCounterparty[chantypes.EventTypeSendPacket] = []uint64{sequence}
+		case chantypes.EventTypeAcknowledgePacket, chantypes.EventTypeTimeoutPacket, chantypes.EventTypeTimeoutPacketOnClose:
+			toDelete[eventType] = []uint64{sequence}
+			toDeleteCounterparty[chantypes.EventTypeRecvPacket] = []uint64{sequence}
+			toDelete[chantypes.EventTypeSendPacket] = []uint64{sequence}
+		}
+		// delete in progress send for this specific message
+		pathEnd.packetProcessing[k].deleteMessages(map[string][]uint64{
+			eventType: {sequence},
+		})
+		// delete all packet flow retention history for this sequence
+		pathEnd.messageCache.PacketFlow[k].DeleteMessages(toDelete)
+		counterparty.messageCache.PacketFlow[k].DeleteMessages(toDeleteCounterparty)
+	}
+
```

Comment:
Could we make this a helper function instead of having this nested function definition? I think it would increase readability/maintainability

Reply:
Done